### DISCUSSION TITLE
Select the correct compiler C/C++ for different parsers

### DISCUSSION THREAD

**Discussion by gabyx** [nvim-treesitter/nvim-treesitter#5716] [General]
**Is your feature request related to a problem? Please describe.**
Currently treesitter does only allow compiling certain parsers with a C compiler.
How does treesitter know, which compiler it should select? For `nickel` that would be a C++ compiler.

**Describe the solution you'd like**
Could this extension somehow make a config for the language to specify the compiler?

```
install.compilers = {
    nickel = require('nvim-treesitter').install.compilers_cpp_defaults -- results in `[ os.getenv('CXX') , ....] list
}
```

**Describe alternatives you've considered**
Not sure? It would be nice that this plugin detects what compiler a parser needs. Not sure. Treesitter is stating it needs a C compiler, but a bit unclear [here](https://tree-sitter.github.io/tree-sitter/creating-parsers).

**Comment by clason** [comment]
We (like upstream) are actively trying to get rid of C++ scanners, for this reason among others. So one compiler must fit all; this is the case for all supported compilers (`gcc`, `clang`, `zig`).

If(!) there was a convention that parsers created and maintained their own makefiles, this would be better, of course, and we would just call those.

**Reply by clason** [reply]
What is your actual issue? This should already Just Work™️.

**Reply by gabyx** [reply]
if I install `nickel` with `CC=$(which clang++) nvim '+TSInstall nickel'`
it works, otherwise it does not. 

So there are much more parsers which use C++. 
Thanks for the answer.

**Reply by clason** [reply]
You should not need to do that. (And there are not _much_ more -- about 10 out of 230+, many of which are niche or not actively maintained anymore.)

**Reply by gabyx** [reply]
Thanks for the nice clarifications!
